File: ios/Plugin/Services/CrashlyticsService.swift
import Foundation
import Capacitor
import FirebaseCrashlytics

class CrashlyticsService {
    private weak var plugin: CAPPlugin?
    
    init(plugin: CAPPlugin) {
        self.plugin = plugin
    }
    
    func crash(_ call: CAPPluginCall) {
        fatalError("Test Crash - This is a test crash triggered by Firebase Kit")
    }
    
    func forceCrash(_ call: CAPPluginCall) {
        let message = call.getString("message") ?? "Forced crash"
        fatalError(message)
    }
    
    func log(_ call: CAPPluginCall) {
        guard let message = call.getString("message") else {
            call.reject("Message is required")
            return
        }
        
        Crashlytics.crashlytics().log(message)
        call.resolve()
    }
    
    func logException(_ call: CAPPluginCall) {
        guard let message = call.getString("message") else {
            call.reject("Message is required")
            return
        }
        
        let code = call.getString("code") ?? "CUSTOM_ERROR"
        let domain = call.getString("domain") ?? "FirebaseKit"
        
        let userInfo: [String: Any] = [
            NSLocalizedDescriptionKey: message
        ]
        
        let error = NSError(domain: domain, code: code.hashValue, userInfo: userInfo)
        Crashlytics.crashlytics().record(error: error)
        
        call.resolve()
    }
    
    func setUserId(_ call: CAPPluginCall) {
        guard let userId = call.getString("userId") else {
            call.reject("User ID is required")
            return
        }
        
        Crashlytics.crashlytics().setUserID(userId)
        call.resolve()
    }
    
    func setCustomKeys(_ call: CAPPluginCall) {
        guard let attributes = call.getObject("attributes") else {
            call.reject("Attributes are required")
            return
        }
        
        let crashlytics = Crashlytics.crashlytics()
        
        for (key, value) in attributes {
            switch value {
            case let stringValue as String:
                crashlytics.setCustomValue(stringValue, forKey: key)
            case let intValue as Int:
                crashlytics.setCustomValue(intValue, forKey: key)
            case let doubleValue as Double:
                crashlytics.setCustomValue(doubleValue, forKey: key)
            case let boolValue as Bool:
                crashlytics.setCustomValue(boolValue, forKey: key)
            default:
                crashlytics.setCustomValue(String(describing: value), forKey: key)
            }
        }
        
        call.resolve()
    }
    
    func setCrashlyticsCollectionEnabled(_ call: CAPPluginCall) {
        guard let enabled = call.getBool("enabled") else {
            call.reject("Enabled parameter is required")
            return
        }
        
        Crashlytics.crashlytics().setCrashlyticsCollectionEnabled(enabled)
        call.resolve()
    }
    
    func isCrashlyticsCollectionEnabled(_ call: CAPPluginCall) {
        let enabled = Crashlytics.crashlytics().isCrashlyticsCollectionEnabled()
        call.resolve(["enabled": enabled])
    }
    
    func deleteUnsentReports(_ call: CAPPluginCall) {
        Crashlytics.crashlytics().deleteUnsentReports()
        call.resolve()
    }
    
    func sendUnsentReports(_ call: CAPPluginCall) {
        Crashlytics.crashlytics().sendUnsentReports()
        call.resolve()
    }
    
    func recordBreadcrumb(_ call: CAPPluginCall) {
        guard let name = call.getString("name") else {
            call.reject("Name is required")
            return
        }
        
        let params = call.getObject("params")
        let breadcrumb = params != nil ? "\(name): \(params!)" : name
        
        Crashlytics.crashlytics().log(breadcrumb)
        call.resolve()
    }
}

File: android/src/main/java/com/ahsanmahmood/capacitor/firebasekit/services/CrashlyticsService.kt
package com.ahsanmahmood.capacitor.firebasekit.services

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.google.firebase.Firebase
import com.google.firebase.crashlytics.FirebaseCrashlytics
import com.google.firebase.crashlytics.crashlytics

class CrashlyticsService(private val plugin: Plugin) {
    
    private val crashlytics: FirebaseCrashlytics by lazy {
        Firebase.crashlytics
    }
    
    fun crash(call: PluginCall) {
        throw RuntimeException("Test Crash - This is a test crash triggered by Firebase Kit")
    }
    
    fun forceCrash(call: PluginCall) {
        val message = call.getString("message") ?: "Forced crash"
        throw RuntimeException(message)
    }
    
    fun log(call: PluginCall) {
        val message = call.getString("message") ?: return
        crashlytics.log(message)
        call.resolve()
    }
    
    fun logException(call: PluginCall) {
        val message = call.getString("message") ?: "Custom exception"
        val code = call.getString("code")
        val domain = call.getString("domain")
        
        val exception = if (code != null || domain != null) {
            Exception("[$domain:$code] $message")
        } else {
            Exception(message)
        }
        
        crashlytics.recordException(exception)
        call.resolve()
    }
    
    fun setUserId(call: PluginCall) {
        val userId = call.getString("userId") ?: return
        crashlytics.setUserId(userId)
        call.resolve()
    }
    
    fun setCustomKeys(call: PluginCall) {
        val attributes = call.getObject("attributes") ?: return
        
        attributes.keys().forEach { key ->
            when (val value = attributes.get(key)) {
                is String -> crashlytics.setCustomKey(key, value)
                is Int -> crashlytics.setCustomKey(key, value)
                is Long -> crashlytics.setCustomKey(key, value)
                is Float -> crashlytics.setCustomKey(key, value)
                is Double -> crashlytics.setCustomKey(key, value)
                is Boolean -> crashlytics.setCustomKey(key, value)
            }
        }
        
        call.resolve()
    }
    
    fun setCrashlyticsCollectionEnabled(call: PluginCall) {
        val enabled = call.getBoolean("enabled") ?: return
        crashlytics.setCrashlyticsCollectionEnabled(enabled)
        call.resolve()
    }
    
    fun isCrashlyticsCollectionEnabled(call: PluginCall) {
        val result = JSObject()
        result.put("enabled", crashlytics.isCrashlyticsCollectionEnabled())
        call.resolve(result)
    }
    
    fun deleteUnsentReports(call: PluginCall) {
        crashlytics.deleteUnsentReports()
        call.resolve()
    }
    
    fun sendUnsentReports(call: PluginCall) {
        crashlytics.sendUnsentReports()
        call.resolve()
    }
    
    fun recordBreadcrumb(call: PluginCall) {
        val name = call.getString("name") ?: return
        val params = call.getObject("params")
        
        val breadcrumb = if (params != null) {
            "$name: $params"
        } else {
            name
        }
        
        crashlytics.log(breadcrumb)
        call.resolve()
    }
}